### ISSUE TITLE
WinMerge will not allow me to open file

### ISSUE THREAD

**Issue by 4321HotRod** [WinMerge/winmerge#592]
FYI
Instead it permanently shows "folder selection." in the 'file name:' bar and when I press open it just opens the file and never selects it for the folder compare within the WinMerge 'dialog.' 
screen shot after i navigate to documents
![2021-01-31 (5)](https://user-images.githubusercontent.com/78318925/106395018-e799e480-63cd-11eb-934f-fda79767073c.png)
screenshot when I press the open button
![2021-01-31 (6)](https://user-images.githubusercontent.com/78318925/106395024-f2547980-63cd-11eb-987c-14e5cc732f5e.png)
instead of selecting that folder to compare in WinMerge it opens the documents folder within the open/select dialog window. 
This is a new version from what I have used before.
Both on Win10 machines about 3 years old and pretty much identical. 
HP Elitedesk 800G, i7 @4770, 16g ram, 1 tb SSD, upgraded to Win10
-space-
**On the one that does NOT WORK: Version 2.16.10.0 X64**
On the one that does work: Version 2.16.2.0 Unicode X64
For now I will uninstall sub version 10 and try another. 
Just thought you might like to know.

**Comment by sdottaka** [comment]
I'm a little confused. Are you trying to open a file or open a folder?

Either way, I couldn't find any difference in behavior between the old version 2.16.2 and the new version 2.16.10.

**Comment by 4321HotRod** [comment]
Thx for responding.   Sorry for the confusion.  

Answer:  I would like to compare two different "Documents" folders.  (Problem with OneDrive changes) 

Referencing the screen captures I posted: 

No: 1 - I am choosing a folder to open by wading thru the folder structure and when I get to "Documents" I highlight that folder and press open.  [Please note that the subfolder structure under the user name is what you see and I select  "Documents"]

[sidebar: I am expecting that WinMerge will 'open' / select that folder to compare.  Instead, it opens that folder inside the open window pane.  I am expecting it to select that folder to compare.  When I do this on previous version it acts as expected: in other words; WinMerge selects that folder and it shows up in the WinMerge window pane in the white 'slot' (I don't know the term for it) that is below "1st File or Folder" and then I go on to choose the "2nd File or Folder"]

No: 2 - This is what it looks like after pressing the open button (I don't know how to capture the cursor) [Please note that the subfolder (inside user\Documents\ list is what you see in 2. ]
[sidebar: when I highlight the "Documents" folder in the previous version, that folder is inserted into the 'slot' under "1st File or Folder" and I repeat for the other "Documents" folder following a different path. ]

In other words: 
I did the same thing in both versions (albeit different computers) and it worked in one version and not the other.  

Here's hoping I cleared up the confusion!  Thx in advance for taking the time!
Follow-up:
I posted to let y'all know my experience. 
For my use, I uninstalled '10' and installed '2' and it worked as expected.  [so I am good to go}
Strange, huh!?!

**Comment by sdottaka** [comment]
Thank you for your explanation.

However, there is no difference between version 2.16.2 and 2.16.10 and it works as you have shown.

I imagine you probably double-clicked the Open button when working with the Open dialog in version 2.16.2.

WinMerge makes it possible to force a folder selection to use the Open file dialog box, which is a common component of Windows that is originally used to open files.

For this reason, it does not behave intuitively as you have shown. The same is true for 2.16.2.

Windows has a dedicated Dialog box for selecting folders, which will give you the behavior you expect. For this reason, I was thinking of using a separate button for file selection and folder selection and using that dedicated Dialog box for folder selection, but I felt that the screen would be messed up, so I stopped.
